core/adapters/multiply.go

```diff
@@ -3,34 +3,38 @@
 package adapters
 
 import (
+	"encoding/json"
 	"fmt"
 	"math/big"
-	"strconv"
 
 	"chainlink/core/store"
 	"chainlink/core/store/models"
-	"chainlink/core/utils"
 )
 
-// Multiplier represents the number to multiply by in Multiply adapter.
-type Multiplier float64
-
-// UnmarshalJSON implements json.Unmarshaler.
-func (m *Multiplier) UnmarshalJSON(input []byte) error {
-	input = utils.RemoveQuotes(input)
-	times, err := strconv.ParseFloat(string(input), 64)
-	if err != nil {
-		return fmt.Errorf("cannot parse into float: %s", input)
-	}
+// Multiply holds the a number to multiply the given value by.
+type Multiply struct {
+	Times *big.Float `json:"-"`
+}
 
-	*m = Multiplier(times)
+type jsonMultiply struct {
+	Times *JSONBigFloat `json:"times"`
+}
 
-	return nil
+// MarshalJSON implements the json.Marshal interface.
+func (ma *Multiply) MarshalJSON() ([]byte, error) {
```

Comment:
Why is this necessary?

Reply:
Its a bit opinionated, it was to keep types to support JSON from bleeding into other parts of the code, which is not a lot.  Putting it another way, are the adapters structures part of the API or fields to be used internally?